<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1356</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QGroupBox {
    border: 1px solid #CCCCCC;
    border-radius: 9px;
    margin-top: 8px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>6</number>
    </property>
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="mainGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>230</width>
        <height>0</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox {
    margin-top: 0px;
}</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QGroupBox" name="subdivGroupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>200</y>
         <width>211</width>
         <height>541</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>5</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>5</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QLabel" name="subdivTitleLabel">
          <property name="text">
           <string>Subdivision Settings</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="presetLabel">
          <property name="text">
           <string>Preset</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="netPresets">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>Pentagon</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Basis</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>G</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="stepsLabel">
          <property name="text">
           <string>Subdivision Steps</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="subdivSteps"/>
        </item>
        <item>
         <widget class="QLabel" name="pointWeightTextLabel">
          <property name="text">
           <string>Point Weight</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="pointWeightSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;In the line segment &lt;/p&gt;&lt;p&gt;a-b-&lt;span style=&quot; font-weight:600;&quot;&gt;c-d&lt;/span&gt;-e-f&lt;/p&gt;&lt;p&gt;this value changes the weights of the points at &lt;span style=&quot; font-weight:600;&quot;&gt;c&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;d &lt;/span&gt;(the edge points).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="middlePointWeightSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;In the line segment &lt;/p&gt;&lt;p&gt;a-&lt;span style=&quot; font-weight:600;&quot;&gt;b&lt;/span&gt;-c&lt;span style=&quot; font-weight:600;&quot;&gt;-&lt;/span&gt;d-&lt;span style=&quot; font-weight:600;&quot;&gt;e&lt;/span&gt;-f&lt;/p&gt;&lt;p&gt;this value change the weights of the points at &lt;span style=&quot; font-weight:600;&quot;&gt;b&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;e&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="outerPointWeightSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;In the line segment &lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;a&lt;/span&gt;-b-&lt;span style=&quot; font-weight:600;&quot;&gt;c-&lt;/span&gt;d-e-&lt;span style=&quot; font-weight:600;&quot;&gt;f&lt;/span&gt;&lt;/p&gt;&lt;p&gt;this value change the weights of the points &lt;span style=&quot; font-weight:600;&quot;&gt;a&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;f &lt;/span&gt;(the end points). &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="minimum">
           <double>0.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="normalWeightTextLabel">
          <property name="text">
           <string>Normal Weight</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="normalWeightSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;In the line segment &lt;/p&gt;&lt;p&gt;a-b-&lt;span style=&quot; font-weight:600;&quot;&gt;c-d&lt;/span&gt;-e-f&lt;/p&gt;&lt;p&gt;this value changes the weights of the normals at &lt;span style=&quot; font-weight:600;&quot;&gt;c&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;d &lt;/span&gt;(the edge points).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="middleNormalWeightSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;In the line segment &lt;/p&gt;&lt;p&gt;a-&lt;span style=&quot; font-weight:600;&quot;&gt;b&lt;/span&gt;-c&lt;span style=&quot; font-weight:600;&quot;&gt;-&lt;/span&gt;d-&lt;span style=&quot; font-weight:600;&quot;&gt;e&lt;/span&gt;-f&lt;/p&gt;&lt;p&gt;this value change the weights of the normals at &lt;span style=&quot; font-weight:600;&quot;&gt;b&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;e&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="outerNormalWeightSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;In the line segment &lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;a&lt;/span&gt;-b-&lt;span style=&quot; font-weight:600;&quot;&gt;c-&lt;/span&gt;d-e-&lt;span style=&quot; font-weight:600;&quot;&gt;f&lt;/span&gt;&lt;/p&gt;&lt;p&gt;this value change the weights of the normals at &lt;span style=&quot; font-weight:600;&quot;&gt;a&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;f &lt;/span&gt;(the end points). &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="minimum">
           <double>0.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="normalizedSolveCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enabling this will try to fit a conic using the unit normal constraint. If this option is disabled, a separated scaling value for each normal is calculated.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Normalized Solve</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="circularNormalsCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Estimate the normals using oscilating circles.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Circular Normals</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="recalculateNormsCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If this option is enabled, the normals will be re-evaluated at every step. If this option is disabled, the vertex points will keep their normals. Any new edge points will obtain the normal of the conic they were sampled from.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Recalculate Normals</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="useEdgeNormalsCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If enabled, uses a ray perpendical to the edge to intersect with the conic and sample the new point from. If disabled, uses the average of the edge point normals.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;In both cases, the ray originates from the middle of the edge.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Use Edge Normals</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="renderGroupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>211</width>
         <height>181</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>5</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>5</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QLabel" name="renderTitleLabel">
          <property name="text">
           <string>Render Settings</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="controlPointsCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Renders the vertex points of the control net.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Control Points</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="controlCurveCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Renders the edges of the control net.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Control Curve</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="visualizeNormalscheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Visualizes the normal directions of each vertex.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Visualize normals</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="outwardNormalsCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flips the normal direction.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Outward Normals</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="closedCheckBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Instead of treating the points as a line segment, enabling this options will treat the points as a closed loop.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Closed Curve</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="MainView" name="mainView"/>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>netPresets</tabstop>
  <tabstop>subdivSteps</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
